core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -50,6 +50,11 @@
             ReplicaRequest extends ReplicationRequest<ReplicaRequest>,
             PrimaryResultT extends ReplicationOperation.PrimaryResult<ReplicaRequest>
         > {
+
+    public static final int WAIT_ON_ACTIVE_SHARDS_NOT_SET = 0;
+    public static final Setting<Integer> WAIT_ON_ACTIVE_SHARDS_SETTING =
```

Comment:
I wonder if we should make this a String setting that allows values such as `all`.